### PR TITLE
Create client-spec.md

### PR DESCRIPTION
**Pull request by tracymiranda**
Migrated from Google doc to Github (with no graphviz, comments moved to issues)

#### Summary
This is the Sigstore client spec doc being moved to github for public collaboration

#### Release Note
NONE

#### Documentation
This migrates the Google doc with the following modifications:

- Add numbering to sections for ease of referencing sections
- Remove graphviz diagram for now
- Migrate todos to github issues

### CODE REVIEW

File: client-spec.md
Comment (chinenyeokafor) [inline client-spec.md:31]: "signer" is used in various sections, such as §2, §2.1.2, §2.1.5, where "client" might be more appropriate based on the terminology and party definitions. It might be helpful if we review these uses or consider redefining the terms for consistency throughout the doc

File: client-spec.md
Comment (chinenyeokafor) [inline client-spec.md:214]: I didn't see a new issue raised for the "§TODO". Not sure if it was intended to be removed or if it’s still pending

Comment (tracymiranda) [reply client-spec.md:214]: Good catch, opened #13 to ensure this gets resolved.

**Comment by haydentherapper** [comment]
FYI, I just added a few comments to the doc we should copy into issues, around timestamping updates. Lemme know if you need pointers to them.

File: client-spec.md
Comment (haydentherapper) [inline client-spec.md:82]: Can we update to https://datatracker.ietf.org/doc/html/rfc6962#section-3.2, as RFC 9162 isn't implemented?

Comment (haydentherapper) [inline client-spec.md:82]: nit: "using the public key from the Certificate Transparency log", not a cert

File: client-spec.md
Comment (haydentherapper) [inline client-spec.md:134]: We should make this a SHOULD and emphasize that this is the recommended default.

File: client-spec.md
Comment (haydentherapper) [inline client-spec.md:193]: Left a comment about this in the doc, but Verifiers shouldn't abort for this or the next step imo. Verifiers should gather as many timestamps as they can successfully validate, and use each when verifying the certificate.

File: client-spec.md
Comment (haydentherapper) [inline client-spec.md:212]: https://datatracker.ietf.org/doc/html/rfc6962#section-3.2 here too

File: client-spec.md
Comment (haydentherapper) [inline client-spec.md:214]: Can we update to 1.3.6.1.4.1.57264.1.8? This is the new extension, same content, just different format

File: client-spec.md
Comment (tracymiranda) [reply client-spec.md]: Done

File: client-spec.md
Comment (tracymiranda) [reply client-spec.md]: Done

File: client-spec.md
Comment (woodruffw) [inline client-spec.md:83]: Copying from the doc, from @segiddins: this bullet has some inaccuracies in it:

* The leaf certificate has SANs, not a subject (the subject is empty, since we only use SANs)
* The leaf's SANs should be checked not against the CSR's subject (which is incomplete and unauthenticated), but against the OIDC credential that *accompanied* the CSR during issuance
* `AuthenticationServiceIdentifier` doesn't get defined anywhere above, so I _think_ this should probably say something like "encodes the appropriate underlying OIDC IdP issuer" since that's what it means 🙂 

I'm happy to send patches for these improvements, either to this PR or as follow-ups once this is merged. Just let me know whatever works best!

Comment (haydentherapper) [reply client-spec.md:83]: @woodruffw can you copy those into an issue? I'd vote we approve and merge the PR as is and iterate.

**Review by haydentherapper — APPROVED** [review]
Approving as is, assuming we will follow up with outstanding issues.

File: client-spec.md
Comment (woodruffw) [reply client-spec.md:83]: Sure, doing now!

Comment (woodruffw) [reply client-spec.md:83]: Opened #14 with the breakout here.

File: client-spec.md
Comment (tracymiranda) [reply client-spec.md:31]: Opened #17 to track and address after we merge this.

File: client-spec.md
Comment (tracymiranda) [reply client-spec.md:83]: FYI: Added #14 to umbrella tracking issue #12

File: client-spec.md
Comment (tracymiranda) [reply client-spec.md:193]: Tracked as #18 (and also linked in top-level issue #12 )

**Review by lukehinds — APPROVED** [review]

**Pull request merged**
